### ISSUE TITLE
Documentation: man pages

### ISSUE THREAD

**Issue by AladW** [AladW/aurutils#76]
- [X] man-pages(7) structure
- [x] convert to .groff
- [X] aurbuild
- [X] aurchain
- [X] aurqueue
- [X] aursearch
- [x] aursift
- [X] aursync
- [X] repofind
- [X] repoman

See also: #1

**Comment by Earnestly** [comment]
Writing manuals in plain groff using the man macros is actually quite simple, it's just another markup language.  Unless you really need some kind of universal markup language to generate documentation for every conceivable need, I'd not bother with the additional dependency.

**Comment by AladW** [comment]
Thanks for the hint and examples. With some experimenting, I added the first man page with 58ecf1f38090288563420d26fecc548b04550f1e

With `groff -Tascii -man` it looks like this:
```
AURSIFT(1)                  General Commands Manual                 AURSIFT(1)
 


NAME
       aursift - filter packages in the Arch Linux repositories

SYNOPSIS
       aursift <operation> [targets]

DESCRIPTION
       Invoking  aursift  involves  specifying an operation with any potential
       options, and targets to operate on. A target is usually a package name,
       or a (space-separated) string of package base and package version. Tar-
       gets can be provided on stdin.

OPERATIONS
       -S  Check if pkgname is in the official Arch  Linux  repositories,  and
           print  it  to  stdout  when  not. Virtual packages, when found, are
           solved and printed to stderr.

       -U  Compare pkgbase pkgver against packages from any configured reposi-
           tory. Names of updated package bases are printed on stdout.

SEE ALSO
       pacsift(1), expac(1)



                                                                    AURSIFT(1)
```

I wasn't too sure on the actual text, so I took `pacman(1)` as a baseline.

**Comment by Earnestly** [comment]
Add an `AUTHORS` section:
```troff
.SH AUTHORS

.MT https://github.com/AladW
Alad Wenter
.ME
```
Change `<operation>` to `-SU` or keep it consistent with the `usage: ...` synopsis.

You could also wrap those lines to 80 columns or so just to make it easier to view since roff is a proper typesetter unlike HTML which doesn't know how to deal with linebreaks.

Otherwise that seems fine to me.

**Comment by AladW** [comment]
For `<operation>`, as the two operations are exclusive I've put each on one line, without braces. Maybe `usage:` needs similar adjustments.

For `.RS`, the following:
```
.B \-S
.RS
Check if
.I pkgname
is in the official Arch Linux repositories, and print it to stdout if
not. Virtual packages, when found, are solved and printed to stderr.
.RE
```
ends up like:
```
       -S
              Check if pkgname is in the official Arch Linux repositories, and print it to stdout
              if not. Virtual packages, when found, are solved and printed to stderr.
```
I'm unsure on the extra break over `.TP`, since I only have a short option (`acid(1)` is similar, though it has `.TP` over `.TP 4`).

I'll continue work on the wording, thanks.

**Comment by Earnestly** [comment]
It doesn't really matter that they're exclusive, or if they are using something more like `usage: aursift [-S target] [-U target]` where you can change the `target` to something more appropriate which you can reference in the description.

The `.RS` and the output is correct.

**Comment by runical** [comment]
OK, so as I promised on the Arch forums, I'm working on the documentation (currently in markdown). I have short man pages for aursync and aurbuild at the moment.

I also have two questions about aursync: 

* The -n flag. It translates to noprepare, so I assume that it somehow stops preparation, but what does that concretely mean in this case?
* The -i flag is mentioned in the help, but does not seem to have an actual function other than matching the *) clause when parsing. Is this intended or is it extraneous?

**Comment by AladW** [comment]
If noprepare is not set, then version checks are done. In practice this means:
+ Take "pkgname pkgver" pairs from the RPC (alphabetically sorted), and pass those on to ~aursift -U~. The result is a list of package names which are updated in the local (or official) repos. Side note: I should rename ~pair_sync~ to ~list_sync~ on that regard.
+ Replace pkgname with pkgbase in this list. You want the build queue to only contain pkgbase - clone dirs are named after pkgbase, and you want to cd to it for later builds.
+ Take the existing pkgbase queue (~queue_base~), and do an inverse match from the above list. While ~queue_base~ is in total order, and the list is not, this doesn't matter as grep keeps the order of the target file, i.e. ~queue_base~. The result is called ~queue_new~.

In short, prepare checks package versions to see which packages already have the same version of the AUR package, or newer. Those that are, are removed from the build queue.

What noprepare does is simply disable this step, by copying the original pkgbase queue (~queue_base~) to ~queue_new~.

About the ~-i~ flag, yeah it's a left-over from #88 (see #94 for a new approach).

**Comment by Earnestly** [comment]
> In short, prepare checks package versions to see which packages already have the same version of the AUR package, or newer. Those that are, are removed from the build queue.

This may have implications for packages that need relinking

**Comment by AladW** [comment]
True enough. The versioning idea introduced with #28, to avoid building packages twice. I'm not sure what would be the best approach here.
